godot-core/src/builtin/collections/packed_array.rs

```diff
@@ -414,29 +435,51 @@ macro_rules! impl_packed_array {
         #[doc = concat!("Creates a `", stringify!($PackedArray), "` from the given slice.")]
         impl From<&[$Element]> for $PackedArray {
             fn from(slice: &[$Element]) -> Self {
-                let mut array = Self::new();
-                let len = slice.len();
-                if len == 0 {
-                    return array;
-                }
-                array.resize(len);
+                let mut array = match slice.len() {
+                    0 => return Self::new(),
+                    len => Self::default_with_size(len),
+                };
                 let ptr = array.ptr_mut(0);
-                for (i, element) in slice.iter().enumerate() {
-                    // SAFETY: The array contains exactly `len` elements, stored contiguously in memory.
-                    unsafe {
-                        // `GString` does not implement `Copy` so we have to call `.clone()`
-                        // here.
-                        *ptr.offset(to_isize(i)) = element.clone();
-                    }
+                // SAFETY: The array contains exactly `len` elements, stored contiguously in memory.
+                unsafe {
+                    std::slice::from_raw_parts_mut(ptr, slice.len()).clone_from_slice(slice);
                 }
                 array
             }
         }
 
+        #[doc = concat!("Creates a `", stringify!($PackedArray), "` from the given Rust array.")]
+        impl<const N: usize> From<[$Element; N]> for $PackedArray {
+            fn from(arr: [$Element; N]) -> Self {
+                let mut ret = match N {
+                    0 => return Self::new(),
+                    len => Self::default_with_size(len),
+                };
```

Comment:
`if` for early exit is still clearer, even if it takes a bit more LoC.

Reply:
Also, could you rename `ret` to `array`?

Reply:
Done